### PR TITLE
[BC-Breaking] Drop pseudo complex support from spectrogram

### PR DESCRIPTION
**Pull request by mthrok**
Following the plan #1337, this PR drops the support for pseudo complex type from `F.spectrogram` and `T.Spectrogram`. It also deprecates the use of `return_complex` argument.

### CODE REVIEW

File: torchaudio/functional/functional.py [hunk 1]
```diff
@@ -77,25 +77,18 @@ def spectrogram(
         onesided (bool, optional): controls whether to return half of results to
             avoid redundancy. Default: ``True``
         return_complex (bool, optional):
```
Comment (nateanl) [inline torchaudio/functional/functional.py:79]:
```suggestion
        return_complex (bool or None, optional):
```

File: torchaudio/transforms.py [hunk 2]
```diff
@@ -65,13 +65,7 @@ class Spectrogram(torch.nn.Module):
         onesided (bool, optional): controls whether to return half of results to
             avoid redundancy (Default: ``True``)
         return_complex (bool, optional):
```
Comment (nateanl) [inline torchaudio/transforms.py:67]:
```suggestion
        return_complex (bool or None, optional):
```

File: torchaudio/transforms.py [hunk 3]
```diff
@@ -108,7 +102,12 @@ def __init__(self,
         self.center = center
         self.pad_mode = pad_mode
         self.onesided = onesided
-        self.return_complex = return_complex
+        if return_complex is not None:
+            warnings.warn(
+                "`return_complex` argument is now deprecated and is not effective."
+                "`torchaudio.transforms.Spectrogram(power=None)` always return tensor with "
```
Comment (nateanl) [inline torchaudio/transforms.py:108]:
```suggestion
                "`torchaudio.transforms.Spectrogram(power=None)` always returns a tensor with "
```

File: torchaudio/functional/functional.py [hunk 4]
```diff
@@ -77,25 +77,18 @@ def spectrogram(
...
     """
-    if power is None and not return_complex:
+    if return_complex is not None:
         warnings.warn(
-            "The use of pseudo complex type in spectrogram is now deprecated."
-            "Please migrate to native complex type by providing `return_complex=True`. "
-            "Please refer to https://github.com/pytorch/audio/issues/1337 "
-            "for more details about torchaudio's plan to migrate to native complex type."
+            "`return_complex` argument is now deprecated and is not effective."
+            "`torchaudio.functional.spectrogram(power=None)` always return tensor with "
```
Comment (nateanl) [inline torchaudio/functional/functional.py:90]:
```suggestion
            "`torchaudio.functional.spectrogram(power=None)` always returns a tensor with "
```

**Review by nateanl — APPROVED** [review]
LGTM! Just need to address the docstring then we can merge it.

**Review by carolineechen — APPROVED** [review]

File: torchaudio/functional/functional.py (hunk 1 above)
Comment (mthrok) [reply torchaudio/functional/functional.py:79]: Here, `None` is used as an indicator to only know if user pass something, so we will keep it as `bool`.

**Pull request merged**